polkadot/zombienet_tests/elastic_scaling/0001-basic-3cores-6s-blocks.zndsl

```diff
@@ -0,0 +1,22 @@
+Description: Test with adder collator using 3 cores and async backing
+Network: ./0001-basic-3cores-6s-blocks.toml
+Creds: config
+
+# Check authority status.
+elastic-validator-0: reports node_roles is 4
+elastic-validator-1: reports node_roles is 4
+elastic-validator-2: reports node_roles is 4
+elastic-validator-3: reports node_roles is 4
+elastic-validator-4: reports node_roles is 4
+
+
+# Register 2 extra cores to this some-parachain.
+elastic-validator-0: js-script ./assign-core.js with "0" return is 0 within 600 seconds
+elastic-validator-0: js-script ./assign-core.js with "1" return is 0 within 600 seconds
+
+# Wait for 10 relay chain blocks 
+elastic-validator-0: reports substrate_block_height{status="best"} is at least 20 within 120 seconds
+
+# Parachain should progress with 3 blocks per relay chain block, so it's reasonable to expect state to be 
+# at least 50, assuming some tolerance 
+some-parachain: log line contains "BlockData { state: 50, add: 2 }" within 10 seconds
```

Comment:
a more reliable test would be based on the parachain block height

Reply:
also, we can only assert this for parachain 2000. we should check that the other parachain is progressing 3 times slower

Reply:
Not really possible with test collators, since they are not based on substrate and have no graphana endpoint for the parachain.

Reply:
> Not really possible with test collators, since they are not based on substrate and have no graphana endpoint for the parachain.

hmm, this test for example: https://github.com/paritytech/polkadot-sdk/blob/d0ebb850ed2cefeb3e4ef8b8e0a16eb7fb6b3f3e/polkadot/zombienet_tests/smoke/0001-parachains-smoke-test.zndsl#L6, https://github.com/paritytech/polkadot-sdk/blob/d0ebb850ed2cefeb3e4ef8b8e0a16eb7fb6b3f3e/polkadot/zombienet_tests/smoke/0001-parachains-smoke-test.toml#L24
uses adder-collator and asserts the block height

Reply:
but that looks like a very old test which isn't being run any more

Reply:
TBH, I am not sure that is correct.

Reply:
For example, querying the adder collator prometheus endpoint gives:
```
# HELP substrate_block_height Block height info of the chain
# TYPE substrate_block_height gauge
substrate_block_height{status="best",chain="rococo_local_testnet"} 47
substrate_block_height{status="finalized",chain="rococo_local_testnet"} 45
substrate_block_height{status="sync_target",chain="rococo_local_testnet"} 47
```

Which is the bundled full node metrics.

Reply:
added test for non-elastic para